app/assets/javascripts/service_management.coffee

```diff
@@ -279,3 +282,9 @@ $(document).ready ->
     # Update the Chooser when the service interface is modified
     $('#service_entry_guid').on 'change', ->
       populateDynamicChooser(serviceOptionAssignmentCollectionsChooser, $(this).val())
+
+  # Clear a user's selection when they try to filter something.  Otherwise,
+  # a user can remove items from the selected box without being able to see
+  # that it is currently selected.
+  $('#to-filter').on 'focus', ->
```

Comment:
(waiting for response before I give my review)

Reply:
I added it to the existing handler of the to filter.  This changed the behavior slightly.  Instead of clearing selections when the user selects the filter box, it now clears selections when they begin typing.  This still accomplishes the goal of preventing the user from having selected items that are invisible due to filtering.  I believe that this change resolves your concerns.